src/lfx/src/lfx/components/agents/altk_agent.py

```diff
@@ -0,0 +1,426 @@
+                    "edit_mode": EditMode.POPOVER,
+                },
+                {
+                    "name": "type",
+                    "display_name": "Type",
+                    "type": "str",
+                    "edit_mode": EditMode.INLINE,
+                    "description": ("Indicate the data type of the output field (e.g., str, int, float, bool, dict)."),
+                    "options": ["str", "int", "float", "bool", "dict"],
+                    "default": "str",
+                },
+                {
+                    "name": "multiple",
+                    "display_name": "As List",
+                    "type": "boolean",
+                    "description": "Set to True if this output field should be a list of the specified type.",
+                    "default": "False",
+                    "edit_mode": EditMode.INLINE,
+                },
+            ],
+        ),
+        *LCToolsAgentComponent._base_inputs,
+        # removed memory inputs from agent component
+        # *memory_inputs,
+        BoolInput(
+            name="add_current_date_tool",
+            display_name="Current Date",
+            advanced=True,
+            info="If true, will add a tool to the agent that returns the current date.",
+            value=True,
+        ),
+        BoolInput(
+            name="enable_post_tool_reflection",
+            display_name="Post Tool JSON Processing",
+            info="If true, it passes the tool output to a json processing (if json) step.",
+            value=True,
+        ),
+        # Post Tool Processing is applied only when the number of characters in the response
+        # exceed the following threshold
+        IntInput(
+            name="response_processing_size_threshold",
+            display_name="Response Processing Size Threshold",
+            value=100,
+            info="Tool output is post-processed only if the response length exceeds a specified character threshold.",
+            advanced=True,
+            show=True,
+        ),
+    ]
+    outputs = [
+        Output(name="response", display_name="Response", method="message_response"),
+    ]
+
+    async def run_agent(
+        self,
+        agent: Runnable | BaseSingleActionAgent | BaseMultiActionAgent | AgentExecutor,
+    ) -> Message:
+        input_dict: dict[str, str | list[BaseMessage]] = {
+            "input": self.input_value.to_lc_message() if isinstance(self.input_value, Message) else self.input_value
+        }
+        user_query = input_dict["input"].content if hasattr(input_dict["input"], "content") else input_dict["input"]
+        if self.enable_post_tool_reflection:
+            wrapped_tools = [
+                PostToolProcessor(
+                    wrapped_tool=tool,
+                    user_query=user_query,
+                    agent=agent,
+                    response_processing_size_threshold=self.response_processing_size_threshold,
+                )
```

Comment:
No need to call `to_lc_message` here because you are instantly using the context. You can pass the `self.input_value.text` or `self.input_value.get_text()`.